openmc/filter.py

```diff
@@ -2301,3 +2301,151 @@ def get_pandas_dataframe(self, data_size, stride, **kwargs):
             {self.short_name.lower(): filter_bins})])
 
         return df
+
+class WeightFilter(Filter):
+    """Bins tally events based on their weights.
+
+    .. versionadded:: 0.12.2
+
+    Parameters
+    ----------
+    bins : Iterable of int
+        A list or iterable of the weights, as double values.
```

Comment:
Since it's python, it's better to say float rather than double. also the type listed here is 'iterable of int' which isn't quite right.